lib/spack/spack/build_systems/autotools.py

```diff
@@ -140,16 +158,46 @@ def patch(self):
 
     def autoreconf(self, spec, prefix):
         """Not needed usually, configure should be already there"""
-        pass
+        # If configure exists nothing needs to be done
+        if os.path.exists(self.configure_abs_path):
+            return
+        # Else try to regenerate it
+        autotools = ['m4', 'autoconf', 'automake', 'libtool']
+        missing = [x for x in autotools if x not in spec]
+        if missing:
+            msg = 'Cannot generate configure: missing dependencies {0}'
+            raise RuntimeError(msg.format(missing))
+        tty.msg('Configure script not found: trying to generate it')
+        tty.warn('*********************************************************')
+        tty.warn('* If the default procedure fails, consider implementing *')
+        tty.warn('*        a custom AUTORECONF phase in the package       *')
+        tty.warn('*********************************************************')
+        with working_dir(self.configure_directory):
+            m = inspect.getmodule(self)
+            m.libtoolize()
+            m.aclocal()
+            m.autoconf()
+            autogen = Executable('./autogen.sh')
+            autogen()
```

Comment:
Thanks for doing some research @alalazo. I would say, take the packages currently in Spack that call autoreconf and see how many of them will work with your new default method. Go by whatever works for most packages.